### PR TITLE
dependency: upgrade TiDB dependency

### LINKED ISSUE #4298: DM: failed to upgrade tidb and parser dependency
### What did you do?

```
go get github.com/pingcap/tidb@master
go get github.com/pingcap/tidb/parser@master
make dm-worker
```

### What did you expect to see?

build successfully

### What did you see instead?

```
make dm-worker
CGO_ENABLED=0 GO111MODULE=on go build  -trimpath  -ldflags '-X "github.com/pingcap/tiflow/pkg/version.ReleaseVersion=v5.4.0-master-dirty" -X "github.com/pingcap/tiflow/pkg/version.BuildTS=2022-01-12 07:54:10" -X "github.com/pingcap/tiflow/pkg/version.GitHash=9261014edd93902d1b0bcb473aec911e80901721" -X "github.com/pingcap/tiflow/pkg/version.GitBranch=master" -X "github.com/pingcap/tiflow/pkg/version.GoVersion=go version go1.16 linux/amd64" -X "github.com/pingcap/tiflow/dm/pkg/utils.ReleaseVersion=v5.4.0-master-dirty" -X "github.com/pingcap/tiflow/dm/pkg/utils.BuildTS=2022-01-12 07:54:10" -X "github.com/pingcap/tiflow/dm/pkg/utils.GitHash=9261014edd93902d1b0bcb473aec911e80901721" -X "github.com/pingcap/tiflow/dm/pkg/utils.GitBranch=master" -X "github.com/pingcap/tiflow/dm/pkg/utils.GoVersion=go version go1.16 linux/amd64"' -o bin/dm-worker ./dm/cmd/dm-worker
# github.com/pingcap/tiflow/dm/pkg/schema
dm/pkg/schema/tracker.go:370:41: too many arguments in call to tr.dom.DDL().CreateTableWithInfo
        have (session.Session, model.CIStr, *model.TableInfo, ddl.OnExist, bool)
        want (sessionctx.Context, model.CIStr, *model.TableInfo, ddl.OnExist)
make: *** [Makefile:254: dm-worker] Error 2
```

### Versions of the cluster

DM version (run `dmctl -V` or `dm-worker -V` or `dm-master -V`):
```console
master
```

### current status of DM cluster (execute `query-status <task-name>` in dmctl)

```console
(paste current status of DM cluster here)
```

### PR DESCRIPTION
**Pull request by GMHDBJD**
### What problem does this PR solve?

Issue Number: close #4298 

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has exported function/method change
 - Has exported variable/fields change
 - Has interface methods change
 - Has persistent data change

Side effects

 - Possible performance regression
 - Increased code complexity
 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation
 - Need to update key monitor metrics in both TiCDC document and official document

### Release note 

```release-note
None
```

### CODE REVIEW

File: dm/pkg/conn/mockdb.go [hunk 1]
```diff
@@ -61,3 +73,106 @@ func InitVersionDB(c *check.C) sqlmock.Sqlmock {
 	}
 	return mock
 }
+
+// TODO: export Config in https://github.com/pingcap/tidb/blob/a8fa29b56d633b1ec843e21cb89131dd4fd601db/br/pkg/mock/mock_cluster.go#L35
```
Comment (Ehco1996) [inline dm/pkg/conn/mockdb.go:77]: why we need copy cluster code here? seems only difference is `Port`?

**Review by lance6716 — APPROVED** [review]

File: dm/pkg/conn/mockdb.go (hunk 1 above)
Comment (GMHDBJD) [reply dm/pkg/conn/mockdb.go:77]: port is not export in tidb repo, so I copy the main logic now. We may change back after 5.4 release or maintain a mock cluster in DM for further use.

Comment (GMHDBJD) [reply dm/pkg/conn/mockdb.go:77]: No, after https://github.com/pingcap/tidb/pull/31167, mock cluster use a random port but not export in DSN(127.0.0.1:/)

**Pull request merged**